homeassistant/components/insteon/api/config.py

```diff
@@ -0,0 +1,277 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from ..const import (
+    CONF_HOUSECODE,
+    CONF_OVERRIDE,
+    CONF_UNITCODE,
+    CONF_X10,
+    DEVICE_ADDRESS,
+    DOMAIN,
+    ID,
+    SIGNAL_ADD_DEVICE_OVERRIDE,
+    SIGNAL_ADD_X10_DEVICE,
+    SIGNAL_REMOVE_DEVICE_OVERRIDE,
+    TYPE,
+)
+from ..schemas import (
+    build_device_override_schema,
+    build_hub_schema,
+    build_plm_manual_schema,
+    build_plm_schema,
+)
+from ..utils import async_get_usb_ports
+
+HUB_V1_SCHEMA = build_hub_schema(hub_version=1)
+HUB_V2_SCHEMA = build_hub_schema(hub_version=2)
+PLM_SCHEMA = build_plm_manual_schema()
+DEVICE_OVERRIDE_SCHEMA = build_device_override_schema()
+OVERRIDE = "override"
+
+
+def get_insteon_config_entry(hass: HomeAssistant) -> ConfigEntry:
+    """Return the Insteon configuration entry."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        return entry
+    raise ValueError("No Insteon configuration exists")
+
+
+def add_x10_device(hass: HomeAssistant, x10_device: dict):
+    """Add an X10 device to the Insteon integration."""
+    config_entry = get_insteon_config_entry(hass)
+    options_config = {**config_entry.options}
+    x10_config = config_entry.options.get(CONF_X10, [])
+
+    # Check that we are not adding an existing device
+    for curr_device in x10_config:
+        if (
+            curr_device[CONF_HOUSECODE] == x10_device[CONF_HOUSECODE]
+            and curr_device[CONF_UNITCODE] == x10_device[CONF_UNITCODE]
+        ):
+            raise ValueError("Duplicate X10 device")
+
+    x10_config.append(x10_device)
+    options_config[CONF_X10] = x10_config
+    hass.config_entries.async_update_entry(entry=config_entry, options=options_config)
+    async_dispatcher_send(hass, SIGNAL_ADD_X10_DEVICE, x10_device)
+
+
+def remove_x10_device(hass: HomeAssistant, housecode: str, unitcode: int):
+    """Remove an X10 device from the config."""
+
+    config_entry = get_insteon_config_entry(hass)
+    new_options = {**config_entry.options}
+    try:
+        new_options.pop(CONF_X10)
+    except KeyError:
+        # There are no X10 devices so nothing to do
+        return
+
+    new_x10 = [
+        existing_device
+        for existing_device in config_entry.options[CONF_X10]
+        if existing_device[CONF_HOUSECODE].lower() != housecode.lower()
+        or existing_device[CONF_UNITCODE] != unitcode
+    ]
+
+    if new_x10:
+        new_options[CONF_X10] = new_x10
+    hass.config_entries.async_update_entry(entry=config_entry, options=new_options)
+
+
+def add_device_overide(hass: HomeAssistant, override: dict[str, str]):
+    """Add an Insteon device override."""
+
+    if has_device_override(hass, Address(override[CONF_ADDRESS])):
+        raise ValueError("Duplicate override")
+
+    config_entry = get_insteon_config_entry(hass)
+    options_config = {**config_entry.options}
```

Comment:
Please use same variable name in the functions, it makes the code easier to read